File: src/main/java/io/scriptor/type/StringResultConverter.java
package io.scriptor.type;

import io.scriptor.http.HTTPResult;
import io.scriptor.http.HTTPResultString;

public class StringResultConverter implements IConverter<String, HTTPResult<?>> {

    @Override
    public HTTPResult<?> from(final String source) {
        return new HTTPResultString(200, "OK", source);
    }
}

File: src/main/java/io/scriptor/type/StringResultConverter.kt
package io.scriptor.type

import io.scriptor.http.result.HTTPResult
import io.scriptor.http.result.HTTPResultString
import io.scriptor.http.result.HTTPResultVoid

class StringResultConverter : IConverter<String, HTTPResult<*>> {

    override fun from(source: String?): HTTPResult<*> {
        if (source == null) {
            return HTTPResultVoid(200, "OK")
        }
        return HTTPResultString(200, "OK", source)
    }
}
